pysmt/smtlib/parser.py

```diff
@@ -532,7 +532,7 @@ def atom(self, token, mgr):
                             if "." in token:
                                 res = mgr.Real(frac)
                             else:
-                                res = mgr.Int(frac.numerator)
+                                res = mgr.Int(int(frac.numerator)) #MG: Shall we use MPZ for ints?
```

Comment:
Why not just passing ```frac.numerator```?
If we are using gmpy, it is an mpz and we avoid the useless conversion, otherwise it is a python integer and we are happy with it...